### PR TITLE
chore: Move to @math.gl/core

### PR DESCRIPTION
**Pull request by ibgreen**
For #5589 

#### Background
- Cleanup before adopting math.gl 8.6-alpha.1
#### Change List
- Replace `math.gl` with `@math.gl.core`

### CODE REVIEW

File: modules/core/package.json [hunk 1]
```diff
@@ -34,9 +34,9 @@
     "@loaders.gl/core": "^3.0.8",
     "@loaders.gl/images": "^3.0.8",
     "@luma.gl/core": "^8.5.10",
-    "@math.gl/web-mercator": "^3.5.4",
+    "@math.gl/core": "^3.5.6",
+    "@math.gl/web-mercator": "^3.5.6",
     "gl-matrix": "^3.0.0",
-    "math.gl": "^3.5.4",
```
[the remark this answers is not in the document]
Comment (Pessimistress) [reply modules/core/package.json:39]: Anyone using `import {Matrix4} from 'math.gl'` will now face the error of trying to import a non-existent package. What is the cost of keeping a reference even if we don't use it in our own code?

[the remark this answers is not in the document]
Comment (ibgreen) [reply modules/core/package.json:39]: `math.gl` is still there: https://github.com/uber-web/math.gl/tree/master/modules/main.

[the remark this answers is not in the document]
Comment (Pessimistress) [reply modules/core/package.json:39]: But it's no longer installed with deck.gl.

[the remark this answers is not in the document]
Comment (ibgreen) [reply modules/core/package.json:39]: Keep it as "bonus dependency" in case someone was depending on it? Sure thing.

**Review by Pessimistress — APPROVED** [review]

**Pull request merged**